### PR TITLE
Dp/sharepoint collector

### PR DESCRIPTION
**Pull request by darshan-patel**
@NileshN @yogendra689 @amardaxini 

Kindly review required changes for SharePoint collector.

### CODE REVIEW

File: app/service/sharepoint_communicator.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+class SharepointCommunicator
+	
+	def initialize(options= {})
```
[the remark this answers is not in the document]
Comment (darshan-patel) [reply app/service/sharepoint_communicator.rb]: Fixed indentation in commit: https://github.com/Course-Master/collector_app/commit/7d8b402ba13c50e294fd1a4d42bfe13330d1c8cc

File: app/service/sharepoint_integration.rb [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      resource_metadata: {
+        images:       [{ url: nil }],
+        title:        entry["Name"],
+        description:  entry["CheckInComment"],
+        url:          entry["public_url"]
+      },
+      additional_metadata: {
+        path_lower:      entry["ServerRelativeUrl"],
+        parent_name:     entry["parent_name"]
+        # size:            link['size'],
```
[the remark this answers is not in the document]
Comment (darshan-patel) [reply app/service/sharepoint_integration.rb]: Removed commented code in commit : 7d8b402

File: app/service/sharepoint_integration.rb [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    @source_id              = @credentials["source_id"]
+    @organization_id        = @credentials["organization_id"]
+
+    @audience_principal_id  = @credentials["audience_principal_id"]
+    @client_id              = @credentials["client_id"]
+    @client_secret          = @credentials["client_secret"]
+    @site_realm             = @credentials["site_realm"]
+    @sharepoint_url         = @credentials["sharepoint_url"]
+    @token                  = JSON.parse(@credentials['access_token'])
+    @token                  = sharepoint_communicator.get_refresh_token
```
[the remark this answers is not in the document]
Comment (darshan-patel) [reply app/service/sharepoint_integration.rb]: @yogendra689 
```
# this is older token stored in the user table inside database
@token                  = JSON.parse(@credentials['access_token'])
# Refreshing older token using refresh token
@token                  = sharepoint_communicator.get_refresh_token
```

[the remark this answers is not in the document]
Comment (yogendra689) [reply app/service/sharepoint_integration.rb]: can you rename it? its confusing

File: app/service/sharepoint_communicator.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+  def get_refresh_token
+  	@token = get_token("refresh_token", "refresh_token=#{@token['refresh_token']}")
+  end
+
+	def get_token(grant_type, token_param)
+		# Remove RestClient and identify way to use Faraday TODO
+		relative_url = "#{@site_realm}/tokens/OAuth/2"
+    params = "grant_type=#{grant_type}&client_id=#{@client_id}@#{@site_realm}&client_secret=#{@client_secret}&redirect_uri=#{redirect_uri}&resource=#{@audience_principal_id}/#{@sharepoint_url}@#{@site_realm}&#{token_param}"
+    @response = RestClient.post "https://accounts.accesscontrol.windows.net/#{@site_realm}/tokens/OAuth/2", params, {content_type: "application/x-www-form-urlencoded"}
```
[the remark this answers is not in the document]
Comment (darshan-patel) [reply app/service/sharepoint_communicator.rb]:
```
def get_token(grant_type, token_param)
    params = {
      "grant_type"    =>  "#{grant_type}",
      "client_id"     =>  "#{@client_id}@#{@site_realm}"
      "client_secret" =>  "#{@client_secret}"
      "redirect_uri"  =>  "#{redirect_uri}"
      "resource"      =>  "#{@audience_principal_id}/#{@sharepoint_url}@#{@site_realm}&#{token_param}"
    }
    base_url = "https://accounts.accesscontrol.windows.net"
    relative_url = "/#{@site_realm}/tokens/OAuth/2"
    headers = {
      content_type: "application/x-www-form-urlencoded"
    }
    
    conn = Faraday.new(base_url)

    # fetch user account details
    @response = conn.get do |req|
      req.url relative_url
      req.headers = headers
      req.params = {}
    end
  end
```

I tried to use `Faraday` as per above code, however it gives parameter not found error.

File: app/service/sharepoint_integration.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (darshan-patel) [reply app/service/sharepoint_integration.rb]: Removed duplication in commit https://github.com/Course-Master/collector_app/commit/446806b2d6bce8eee0e13998bdfc3f65a5d0041f

**Comment by darshan-patel** [comment]
Closing this PR as new PR created with use of Microsoft Graph API

**Pull request closed without merging**